lib/nerve/reporter/zookeeper.rb

```diff
@@ -62,15 +64,41 @@ def stop()
     end
 
     def report_up()
-      zk_save
+      if not @zk.connected?
+        log.error "nerve: error in reporting up on zk node #{@full_key}: loss connection"
+      else
+        begin
+          zk_save
+        rescue *ZK_CONNETION_ERRORS => e
+          log.error "nerve: error in reporting up on zk node #{@full_key}: #{e.message}"
```

Comment:
do we want to swallow the errors? what would be returned in that case?

Reply:
nothing is returned in the case of connection error. we could potentially have re-try logic. but the loop in service_watcher will do the same essentially.

Reply:
actually, let me add a dd metric here.